src/__tests__/useQuery-test.tsx

```diff
@@ -101,310 +103,490 @@ const FILTERED_TASKS_QUERY = gql`
   }
 `;
 
+const linkReturningError = new ApolloLink(
+  () =>
+    new Observable(observer => {
+      observer.error(new Error('Simulating network error'));
+    })
+);
+
+function createMockClient(link?: ApolloLink) {
+  return createClient({ link, mocks: TASKS_MOCKS });
+}
+
+interface TasksProps<TVariables = any> extends QueryHookOptions<TVariables> {
+  query: DocumentNode;
+}
+
 function TaskList({ tasks }: { tasks: Array<{ id: number; text: string }> }) {
   return (
-    <ul data-testid="task-list">
+    <ul>
       {tasks.map(task => (
         <li key={task.id}>{task.text}</li>
       ))}
     </ul>
   );
 }
 
-function TasksLoader({
-  query,
-  ...restOptions
-}: { query: DocumentNode } & QueryHookOptions<any>) {
-  const { data, error } = useQuery(query, restOptions);
-  if (error) {
-    throw error;
-  }
-
-  return <TaskList tasks={data.tasks} />;
-}
-
-function TasksLoaderWithoutSuspense({
-  query,
-  ...restOptions
-}: { query: DocumentNode } & QueryHookOptions<any>): ReactElement<object> {
-  const { data, error, errors, loading } = useQuery(query, {
-    ...restOptions,
-    suspend: false,
-  });
+function Tasks({ query, ...options }: TasksProps) {
+  const { data, error, errors, loading } = useQuery(query, options);
 
   if (error) {
-    throw error;
+    return <>{error.message}</>;
   }
+
   if (errors) {
-    throw new Error('Errors');
+    return (
+      <>
+        {errors.map(x => (
+          <Fragment key={x.message}>{x.message}</Fragment>
+        ))}
+      </>
+    );
   }
+
   if (loading) {
     return <>Loading without suspense</>;
   }
+
+  if (!data) {
+    return <>Skipped loading of data</>;
+  }
+
   return <TaskList tasks={data.tasks} />;
 }
 
-afterEach(cleanup);
+interface TasksWrapperProps extends TasksProps {
+  client: ApolloClient<object>;
+}
 
-it('should return the query data', async () => {
-  const client = createClient({ mocks: TASKS_MOCKS });
-  const { container } = render(
+function TasksWrapper({ client, ...props }: TasksWrapperProps) {
+  return (
     <ApolloProvider client={client}>
-      <Suspense fallback={<div>Loading</div>}>
-        <TasksLoader query={TASKS_QUERY} />
+      <Suspense fallback={<>Loading</>}>
```

Comment:
I don't like `<Suspense>` is always used, even for test where `{ suspend: false }` is used. To fix that and minimize DRY we could:

1. Create a helper component:

```typescript
const SuspenseCompat: React.SFC<{}> = ({ children }) => <>{children}</>;
```

2. Add the following check inside `TasksWrapper`:

```typescript
  const SuspenseComponent = props.suspend !== false ? Suspense : SuspenseCompat;
```

3. Replace usage of suspense inside of `TasksWrapper` with `SuspenseComponent`.